substra/sdk/client.py

```diff
@@ -949,3 +951,111 @@ def organization_info(self) -> models.OrganizationInfo:
     def cancel_compute_plan(self, key: str) -> None:
         """Cancel execution of compute plan. Nothing is returned by this method"""
         self._backend.cancel_compute_plan(key)
+
+    @logit
+    def wait_compute_plan(
+        self, key: str, *, timeout: Optional[float] = None, polling_period: float = 2.0, raises: bool = True
+    ) -> models.ComputePlan:
+        """Block the execution until the compute plan finishes.
+
+        It is considered finished when the status is done, failed or cancelled.
+        If a timeout is defined, will raises a `FutureTimeoutError` once it reached it.
+
+        Args:
+            key (str): the key of the compute plan to wait
+            timeout (Optional[float]): maximum time to wait. If set to None, will hang until completion.
+            polling_period (float): time to wait between to checks, in seconds. Default to 2.0.
+            raises (bool): wether or not to raises exception if the execution fails. Default to True.
```

Comment:
typo and also some modification to make it "sound more english" though I'm not a native speaker so..

```suggestion
            key (str): the key of the compute plan to wait for
            timeout (Optional[float]): maximum time to wait. If set to None, will hang until completion.
            polling_period (float): time to wait between to checks, in seconds. Defaults to 2.0.
            raises (bool): whether to raise an exception if the execution fails. Defaults to True.
```